### DISCUSSION TITLE
nodon SIN-4-1-20, switch operation timeout / error

### DISCUSSION THREAD

**Discussion by git-greatguy** [Koenkk/zigbee2mqtt#15683] [Devices]
Hi,

Installed a nodon [SIN-4-1-20] this morning. Pairing with Z2m without problem.
However, cant control switch relay.
Here is what i see so far trying to get it working : 

as described in documentation, pushlished mqtt message
```
zigbee2mqtt/[INT] Resistance Ballon ECS/set
{"state": "OFF"}
```

timeout reported in z2m

` 2022-12-21 14:56:01Publish 'set' 'state' to '[INT] Resistance Ballon ECS' failed: 'Error: Command 0x9035eafffed8d50e/242 genOnOff.off({}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Timeout - 62107 - 242 - 17 - 6 - 11 after 10000ms)'`

tried to send "genonoff" command through z2m dev console

`2022-12-21 14:57:36Publish 'set' 'write' to '[INT] Resistance Ballon ECS' failed: 'Error: Write 0x9035eafffed8d50e/1 genOnOff({"onOff":1}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'READ_ONLY')'`

Read only ??

tried, to reset, pair again etc.. cant control through z2m.

FYI, connectivity is good, as i switch manually on / off relay with physical button on module, z
2m get instantly updated value on / off

any idea what's wrong with this zigbbee module ?

**Comment by git-greatguy** [comment]
ok got it working ! "Answer" was in my first post when timeout refer to endpoind 242.

i stopped z2m and edit database.db

these parts :

change from 
`"epList":[242,1] `
to 
`"epList":[1,242]`

and 
`"endpoints":{"242":{"profId":41440,"epId":242,"devId":102, `
to 
`"endpoints":{"1":{"profId":41440,"epId":1,"devId":102,`

i dont understand what endpoint 242 refer to ?

**Reply by euidzero** [reply]
Thx a lot. Worked for me too.

**Reply by MichaelMure** [reply]
How did you even figure that out? Is that a zigbee2mqtt bug?

**Reply by MichaelMure** [reply]
I just did the first change (`[242,1]` --> `[1,242]`) and it worked for me.

**Reply by MichaelMure** [reply]
https://github.com/Koenkk/zigbee2mqtt/issues/15962

**Reply by tifriis208** [reply]
A big thanks for your help, got this issue yesterday by switching from ZHA to zigbee2mqtt. So it still a bug ?

For people wondering how to modify the database:
- I used the addon " Advanced SSH & Web Terminal"
- `cd /config/zigbee2mqtt`
- I maked a copy of database before (just in case) : `cp database.db database.db.old`
- open and edit the databse: `nano databse.db` and look for the line with the ID of your Nodon, then change [242,1] to [1,242]
- from the addon page of home assistant, go to zigbee2mqtt, then restart the addon

**Comment by morcus** [comment]
Thank you. I had the same problem with the Adeo (nodon) SIN-4-1-21_EQU module and changing [242,1] to [1,242] fixed the issue.

**Comment by barbierp35** [comment]
Hi,
I use Zigbee2Mqtt in an HomeAssistant installation and I have exactly the same issue for my `SIN-4-FP-21_EQU` module : 
```
2025-07-29 13:45:16z2m: Publish 'set' 'state' to 'Chauffage chambre' failed: 'Error: ZCL command 0x4c5bb3fffe458994/242 genOnOff.on({}, {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"reservedBits":0,"writeUndiv":false}) failed (Timeout - 37648 - 242 - 91 - 6 - 11 after 10000ms)'
```

I want to test your fix with changing 1 and 242, but I don't find the database.db file in my Home assistant.

@git-greatguy @euidzero @MichaelMure or @morcus can you help me for this please ?
Thank you

**Reply by barbierp35** [reply]
Finally I found database.db file with a ssh connection to my HA server. The file is just hide in the VsCode home assistant addon. I will test your fix, thank you.

**Comment by smuteau** [comment]
Hi evryone, 
I just paired today without problem one SIN-4-1-20 and I got issue to control ON/OFF through z2m UI (z2m 2.7.1).
I have installed antoher one nearly one year ago without problem (maybe 1.4.2, don't remember exactly).

For this one, I had to modify `database.db` and change from `"epList":[242,1]` to `"epList":[1,242]` as explain here.

But my z2m didn't work yet because of this error message at container's start :
`MQTT retention requires protocol version 5`
I found some info about this [here](https://github.com/Koenkk/zigbee2mqtt/discussions/19886).

So I modified in `configuration.yaml` version of MQTT from 4 to 5 : 
```
mqtt:
  base_topic: zigbee2mqtt
  server: mqtt://localhost:1885
  user: xx
  password: xx
  keepalive: 60
  reject_unauthorized: true
  version: 5
```
And then z2m started to work well again !
That was my journey of 31st december to be ready for 2026 ;)
